### PR TITLE
[FeedManager][FeedStorage] Ensures particular crop-field pair maps to only one storage

### LINKED ISSUE #2631: [Feed][Storage] Prevent multiple storages mapping to the same crop-field pairs.
**Is your feature request related to a problem? Please describe.**
Currently in the Feed Storage module the following is true:
1. A single crop harvested from a single field is mapped to a Feed storage that produces a single rufas feed.
2. A user has the ability to create 2+ storages that map the same harvested crop from the same field where each produces a different single rufas feed.
3. There is no way for the user to map particular harvests or portions of harvests from the same field to separate storages.

**Describe the solution you'd like**
There should be a restriction put in place that when creating a feed storage, there can only be one mapping of the same crop grown in the same field. If there are 2 storages that map to the same crop-field combo, we need to raise an error.

**Describe alternatives you've considered**
Somewhere down the line, we may want part 2 from above to be true, but that is not on our short term goals' list.

### PR DESCRIPTION
**Pull request by ew3361zh**
Prevents crop-field pairings from mapping to multiple storages.

### Context

Issue(s) closed by this pull request: closes #2631

### What

1. Extracts function to validate uniqueness of storage names for clarity.
2. Creates helper function to validate that the storage config does not have multiple storage configs that map the same crop-field pairings to different storages.

### Why

Mapping the same crop-field pairing to multiple storages is not a desired behavior right now.

### How

The new helper function `_validate_storage_config_names()` creates a list of tuples of each storage's `crop_name` and `field_name` for the storage configs selected by the user and checks that there is no duplication of any combination of those two between any of the selected storages.

### Test plan

Alter the storage configs to have 2 storages that contain the same crop-name and field name mapping and ensure that both storages are in the list of storage configs to be used in the simulation. This should fail.

### Input Changes

N/A

### Output Changes

- N/A

#### Filter

```json

```

### CODE REVIEW

File: RUFAS/biophysical/feed_storage/feed_manager.py [hunk 1]
```diff
@@ -170,6 +166,44 @@ def _create_all_storages(
...
                 )
 
+    def _validate_storage_config_names(self, all_configs: list[dict[str, Any]]) -> None:
+        """Validates that all storage configuration names are unique."""
+        storage_name_counts = Counter(storage_config.get("name") for storage_config in all_configs)
+        duplicate_names = [name for name, count in storage_name_counts.items() if count > 1]
+        if duplicate_names:
+            raise ValueError(
+                f"Duplicate storage config names found: {duplicate_names}. Each storage config must have a unique name."
+            )
```
Comment (allisterakun) [inline RUFAS/biophysical/feed_storage/feed_manager.py:185]: Do we also want to report to the OM?

File: RUFAS/biophysical/feed_storage/feed_manager.py [hunk 2]
```diff
@@ -170,6 +166,44 @@ def _create_all_storages(
...
+                f"Combination {combo} used by storages: {', '.join(sorted(names))}"
+                for combo, names in duplicate_details.items()
+            ]
+            details = "\n".join(detail_lines)
+
+            raise ValueError(
+                "Duplicate (crop_name, field_name) combinations found:\n"
+                f"{details}\n"
+                "Each combination must be unique across all storage configurations."
+            )
```
Comment (allisterakun) [inline RUFAS/biophysical/feed_storage/feed_manager.py:225]: Same as above: Do we also want to report to the OM?

**Review by allisterakun — APPROVED** [review]
LGTM!

File: tests/test_biophysical/test_feed_storage/test_feed_manager.py [hunk 3]
```diff
@@ -294,6 +294,89 @@ def test_create_all_storages(
...
+        {"name": "S2", "other": 2},
+        {"name": "S3", "other": 3},
+    ]
+    add_error = mocker.patch.object(feed_manager._om, "add_error")
+
+    feed_manager._validate_storage_config_names(configs)
+    add_error.assert_not_called()
+
+
+def test_validate_storage_config_names_duplicate_raises(feed_manager: FeedManager, mocker: MockerFixture) -> None:
```
Comment (matthew7838) [inline tests/test_biophysical/test_feed_storage/test_feed_manager.py:375]: I would suggest adding a test for unique "name" and duplicate "other". It's fine to be a separate test and or using parameterize.

**Review by matthew7838 — APPROVED** [review]
LGTM, left a minor suggestion on testing.
